app/src/main/java/com/github/se/orator/ui/offline/OfflineRecordingScreen.kt

```diff
@@ -4,93 +4,140 @@ import android.Manifest
+import com.github.se.orator.R
+import com.github.se.orator.model.symblAi.SpeakingViewModel
 import com.github.se.orator.ui.navigation.NavigationActions
-import com.github.se.orator.ui.speaking.OfflineSpeakingViewModel
+import com.github.se.orator.ui.navigation.Screen
 
-/**
- * OfflineRecordingScreen is a composable screen for offline mode. It allows the user to record
- * audio locally and play it back.
- *
- * @param navigationActions Provides navigation actions for navigating between screens.
- * @param question The question text selected by the user for offline practice.
- * @param offlineSpeakingViewModel ViewModel for managing audio recording and playback
- *   functionality.
- */
 @SuppressLint("MissingPermission")
 @Composable
 fun OfflineRecordingScreen(
     navigationActions: NavigationActions,
-    question: String, // Accept question as a parameter
-    offlineSpeakingViewModel: OfflineSpeakingViewModel =
-        viewModel(factory = OfflineSpeakingViewModelFactory(LocalContext.current))
+    question: String,
+    speakingViewModel: SpeakingViewModel = viewModel()
 ) {
-  // Local state to track whether recording is in progress
-  var isRecording by remember { mutableStateOf(false) }
-
-  // Local state to track if the recording permission has been granted
   var permissionGranted by remember { mutableStateOf(false) }
-
-  // Set up permission launcher for requesting microphone access
   val permissionLauncher =
-      rememberLauncherForActivityResult(
-          contract = ActivityResultContracts.RequestPermission(),
-          onResult = { isGranted -> permissionGranted = isGranted })
+      rememberLauncherForActivityResult(contract = ActivityResultContracts.RequestPermission()) {
+          isGranted ->
+        permissionGranted = isGranted
+        if (isGranted) {
+          speakingViewModel.onMicButtonClicked(true)
+        }
+      }
 
-  // Request recording permission on initial composition
-  LaunchedEffect(Unit) { permissionLauncher.launch(Manifest.permission.RECORD_AUDIO) }
+  val isRecording by speakingViewModel.isRecording.collectAsState()
+  val infiniteTransition = rememberInfiniteTransition()
+  val scale by
+      infiniteTransition.animateFloat(
+          initialValue = 1f,
+          targetValue = 1.3f,
+          animationSpec =
+              infiniteRepeatable(
+                  animation = tween(600, easing = LinearEasing), repeatMode = RepeatMode.Reverse))
+
+  DisposableEffect(Unit) {
+    onDispose {
+      if (isRecording) {
+        speakingViewModel.endAndSave()
+      }
+    }
+  }
+  val colors = MaterialTheme.colorScheme
 
-  // Layout for the offline speaking screen
   Column(
-      modifier = Modifier.fillMaxSize().padding(16.dp),
-      verticalArrangement = Arrangement.Center,
+      modifier =
+          Modifier.fillMaxSize()
+              .background(colors.background)
+              .padding(WindowInsets.systemBars.asPaddingValues())
+              .padding(horizontal = 16.dp)
```

Comment:
Avoid Hardcoded UI Values: I noticed that some UI elements are using hardcoded padding and sizing values. To ensure a consistent and responsive design across different screen sizes, consider using the createAppDimensions function. This approach allows padding and dimensions to adapt based on the device's screen dimensions, enhancing the overall user experience by providing a more adaptable and visually appealing interface.